Adding Databricks service for Redpanda + Filesystem catalog use case

rpdevmp opened the pull request:
This PR adds a Databricks service file for a specific use case:
Redpanda + Filesystem catalog use case, similar to BigQuery, where Databricks is used as a query engine

This PR doesn't include Databricks Managed Tables / non-filesystem based REST catalog for Databricks. That would be a separate PR, once we have preview feature (managed tables) enabled.

## Backports Required

- [x ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

* none

rpdevmp committed:
Adding Databricks service for Redpanda + Filesystem catalog use case

rpdevmp commented:
More details on use cases and setup could be found at:
https://redpandadata.atlassian.net/wiki/spaces/DEVPROD/pages/1106640902/Databricks

dotnwat commented on tests/rptest/services/databricks_service.py:
```diff
@@ -0,0 +1,434 @@
+from pyspark.sql import SparkSession
+
+
+class DatabricksService:
```
I'm curious if this is a "service" (in the sense that a binary/server is running and managed by ducktape) or if this is more like a client to the databricks service?

rpdevmp replied:
You're correct, Noah
This is a client for a hosted Databricks service rather than a ducktape-managed service process.. I will push another commit to update the docs note and clarify

rpdevmp replied:
Updated

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--b8aae581-802a-4817-9765-1c485920af32-->
<!--TEST RESULTS-->
<!--5cebfd01-3807-4866-8d62-e6b3e0129185-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62804">build#62804</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-----------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| distributed_kv_stm_tests_rpunit.distributed_kv_stm_tests_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/62804#01957314-1f4d-4222-82f8-43033e593d89 | FLAKY | 1/2 |
| rptest.tests.compaction_recovery_test.CompactionRecoveryUpgradeTest.test_index_recovery_after_upgrade | ducktape | https://buildkite.com/redpanda/redpanda/builds/62804#0195735b-f4af-41cb-9612-41a5e6efb908 | FLAKY | 1/2 |
| rptest.tests.datalake.datalake_e2e_test.DatalakeE2ETests.test_topic_lifecycle.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/62804#01957370-73d7-413f-a49e-43b254dad1a5 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62882">build#62882</a></summary>


| test_id                                                                                     | test_kind   | job_url                                                                                   | test_status   | passed   |
|:--------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.consumer_group_test.ConsumerGroupTest.test_group_lag_metrics                   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62882#01958267-768b-47d1-a873-d798da9b574d | FLAKY         | 1/3      |
| rptest.transactions.consumer_offsets_test.VerifyConsumerOffsets.test_consumer_group_offsets | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62882#0195824c-0fbc-4a4e-9577-0b6c1549e4aa | FLAKY         | 1/2      |
</details>
<!--/body-->

nvartolomei commented:
How credentials are passed to the client? I.e. how do I use this client on my local machine with ducktape task?

rpdevmp committed:
Added environment variables check and updated client describtion

rpdevmp commented:
> How credentials are passed to the client? I.e. how do I use this client on my local machine with ducktape task?

Hosted Databricks service requires these environment variables to be set:
"DATABRICKS_HOST" and "DATABRICKS_TOKEN"

I updated the code to check for these variables and raise an error if they were not set.

nvartolomei commented:
@rpdevmp would it be fair to ask you to add relevant changes to vtools too so these are propagated in some form to the docker container to facilitate local testing?

nvartolomei commented:
Btw, the code is gone from this PR.

rpdevmp committed:
Renamed the file to databricks_client.py

rpdevmp commented:
> @rpdevmp would it be fair to ask you to add relevant changes to vtools too so these are propagated in some form to the docker container to facilitate local testing?

Sure, I will ping you in DM about vtools part. I renamed the file as _client and added it. Thank you for the reviews.

dotnwat commented on tests/rptest/services/databricks_client.py:
```diff
@@ -0,0 +1,452 @@
+from pyspark.sql import SparkSession
+
+
+class DatabricksClient:
```
maybe it this should be moved to `tests/rptest/clients/...` instead of `.../services/`?

rpdevmp committed:
Moved under clients

dotnwat commented:
@rpdevmp is this ready for review?

nvartolomei commented:
I'm closing (cleaning up the review queue) this as the work is superseded by https://github.com/redpanda-data/redpanda/pull/25339. Let me know if that's not the case and we still want to merge parts of this.
